services/context-box/server/domain/usecases/enqueue_configs.go

```diff
@@ -0,0 +1,215 @@
+		// If item is already in some queue (scheduler / builder) then skip and move to the next item
+		if u.builderQueue.Contains(configInfo) || u.schedulerQueue.Contains(configInfo) {
+			continue
+		}
+
+		// Initially when the config is received from the frontend microservice, the desired state of the config is not built,
+		// due to which the DsChecksum will be nil.
+		if !utils.Equal(configInfo.DsChecksum, configInfo.MsChecksum) {
+			// If scheduler TTL is <= 0 AND config has no errorMessage, add item to the scheduler queue
+			if configInfo.SchedulerTTL <= 0 && !configInfo.HasError() {
+				if err := u.DB.UpdateSchedulerTTL(configInfo.Name, defaultSchedulerTTL); err != nil {
+					return err
+				}
+
+				// The item is put in the scheduler queue. The scheduler microservice will eventually pull the corresponding
+				// config and build its desired state
+				u.schedulerQueue.Enqueue(configInfo)
+				configInfo.SchedulerTTL = defaultSchedulerTTL
+
+				continue
+			} else if !configInfo.HasError() {
+				// If the item is already present in the scheduler queue but the config is still not pulled by the scheduler
+				// microservice, then reduce its scheduler TTL by 1.
+				configInfo.SchedulerTTL = configInfo.SchedulerTTL - 1
+			}
+		}
+
+		// After the config has its desired state built, the infrastructure needs to be provisioned. The DsChecksum and CsChecksum
+		// doesn't match since the infrastructure is not built yet.
+		if !utils.Equal(configInfo.DsChecksum, configInfo.CsChecksum) {
+			// If builder TTL <= 0 AND config has no errorMessage, add item to the builder queue
+			if configInfo.BuilderTTL <= 0 {
+				// If no BUILD error OR if triggered for deletion in builder microservice.
+				if !configInfo.HasError() || configInfo.scheduleDeletion() {
+					if err := u.DB.UpdateBuilderTTL(configInfo.Name, defaultBuilderTTL); err != nil {
+						return err
+					}
+
+					// The item is put in the builder queue. The builder microservice will eventually pull the corresponding
+					// config and provision the corresponding infrastructure.
+					u.builderQueue.Enqueue(configInfo)
+					configInfo.BuilderTTL = defaultBuilderTTL
+
+					continue
+				}
+			} else if !configInfo.HasError() {
+				// If the item is already present in the builder queue but the config is still not pulled by the builder
+				// microservice, then reduce its scheduler TTL by 1.
+				configInfo.BuilderTTL = configInfo.BuilderTTL - 1
+			}
+		}
+
+		// save data if both TTL were subtracted
+		if err := u.DB.UpdateSchedulerTTL(configInfo.Name, configInfo.SchedulerTTL); err != nil {
+			return nil
+		}
+		if err := u.DB.UpdateBuilderTTL(configInfo.Name, configInfo.BuilderTTL); err != nil {
+			return nil
+		}
+	}
+
+	return nil
+}
+
+// Fetches all configAsBSON from MongoDB and converts each configAsBSON to ConfigInfo
+// Then returns the list
+func getConfigInfosFromDB(mongoDB ports.DBPort) ([]*ConfigInfo, error) {
+	configAsBSONList, err := mongoDB.GetAllConfigs()
```

Comment:
Are you sure the data returned is BSON? I think its proto a message slice, so please rename to `dbConfigs` or something like that. (not `List` as the returned value is slice)